### ISSUE TITLE
read.argo() problem (likely caused by encoding changes)

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1993]
I get an error in argo.R#1652 because `trimString()` cannot handle a particular string in a particular Argo file.  (Not all files have this problem ... I'm reading about 1500 files and this occurs partway through.)
```R
a <- read.argo("/Users/kelley/data/argo/D4900883_003.nc", debug=1)
```
yields
```
Error in gsub(" *$", "", s) : input string 9 is invalid
In addition: Warning message:
In gsub(" *$", "", s) :
 unable to translate 'r=1.000401, <b1> 2.177525e-005    
```

I've attached the file.
[D4900883_003.nc.gz](https://github.com/dankelley/oce/files/9270666/D4900883_003.nc.gz)

**Comment by dankelley** [comment]
The problem goes away if I use `base::trimws()` instead of `oce::trimString()`.

I can see why: `trimws` is using `perl=TRUE` in a `sub()` call.

I think I'll go through all of oce, changing `trimString()` to `trimws()`, since I wrote the former without being aware that I was essentially duplicating (badly) the latter.

**Comment by dankelley** [comment]
Wow, the only place I even use `trimString()` is within `argo.R`.

```
➜  R git:(develop) ✗ git grep -n trimString
argo.R:1265:        as.vector(trimString(ncdf4::ncvar_get(file, maybeLC("PLATFORM_NUMBER", lc)))) else NULL
argo.R:1278:    ###         as.vector(trimString(ncdf4::ncvar_get(file, maybeLC("FLOAT_SERIAL_NO", lc)))) else NULL
argo.R:1284:        as.vector(trimString(ncdf4::ncvar_get(file, maybeLC("PROJECT_NAME", lc)))) else NULL
argo.R:1289:        as.vector(trimString(ncdf4::ncvar_get(file, maybeLC("PI_NAME", lc)))) else NULL
argo.R:1297:        res@metadata$stationParameters <- trimString(ncdf4::ncvar_get(file, maybeLC("STATION_PARAMETERS", lc)))
argo.R:1347:        as.vector(trimString(ncdf4::ncvar_get(file, maybeLC("DC_REFERENCE", lc)))) else NULL
argo.R:1362:        as.vector(trimString(ncdf4::ncvar_get(file, maybeLC("INST_REFERENCE", lc)))) else NULL
argo.R:1367:        as.vector(trimString(ncdf4::ncvar_get(file, maybeLC("FIRMWARE_VERSION", lc)))) else NULL
argo.R:1372:        as.vector(trimString(ncdf4::ncvar_get(file, maybeLC("WMO_INST_TYPE", lc)))) else NULL
argo.R:1434:        as.vector(trimString(ncdf4::ncvar_get(file, maybeLC("POSITIONING_SYSTEM", lc)))) else NULL
misc.R:5467:trimString <- function(s)
```

**Comment by dankelley** [comment]
While I'm rebuilding oce, I decided to look at the file.  The bad line is
```
"r=1.000401, \261 2.177525e-005" ;
```
where I've trimmed a whack of trailing whitespace to highlight the gist (after the comma).

I don't quite know what `ncdump` means by that, but if it's hexadecimal, it means this is an a with "ogonek", whatever the heck that means. https://www.codetable.net/decimal/261

But this could just be what `ncdump` is doing.  Maybe that is a hex code, in which case it is a "latin small leter script G" (https://www.codetable.net/hex/261).

**Comment by dankelley** [comment]
This error does not occur with the R-4.2.1 (the present-day R) but I do see it in the R-devel that I built up a few weeks ago (which I think has been reverted on CRAN, but my guess is that it will come back in some form, later this year). 

Therefore I've decided to make it give a warning, as below
```R
> a <- read.argo("/Users/kelley/data/argo/D4900883_003.nc")
Warning message:
In read.argo("/Users/kelley/data/argo/D4900883_003.nc") :
  cannot trim whitespace from a value for metadata$scientificCalibCoefficient
```

**Comment by dankelley** [comment]
Solved in "develop" commit 073da7ae53d5f1b087e4dc807272654fce6fde22.  Test on 1087 argo files (south of Nova Scotia); there were 4 files with this problem, always in the field named `metadata@scientificCalibCoefficient .  I don't know or care what the character is.  All `read.argo()` does now is to issue a warning, and keep the field as-is, without attempting to trim leading and trailing spaces from this field.  Users may know what to do, although I have never used this field in any work.
